anvio/genomedistance.py

```diff
@@ -65,8 +64,11 @@ def are_redundant(self, genome1, genome2):
         return False
 
     def dereplicate(self):
-        for genome1 in self.genome_names:
-            for genome2 in self.genome_names:
+        names = list(self.genome_names)
+        for i in range(len(names))
```

Comment:
by looking at this very quickly, these for statements seem to be missing `:` characters at the end of them ;)

plus, the next line also has an additional syntax error I think (`i+`).

Reply:
more generally, I think there must be a much more pythonic way to do this part. Whenever one sees variable names `i` and `j` they should know that they are doing something not so optimal :) For instance, something like this would have given all permutations of genome names in the `self.genome_names` list:

``` python
(...)
from itertools import permutations 

(...)

genome_pairs = permutations(self.genome_names, 2) 
```

and would remove the need for those nested `for` loops.

Reply:
Thank you! I'll take care of this